### PR TITLE
Create volunteer posting card component

### LINKED ISSUE #82: Volunteer Posting Card
### Summary and Background
Volunteers can view a condensed card outlining basic info for a posting. They can then choose to view more details on the posting or submit/edit availabilities. To be used in the 'Browse Volunteer Postings' page.

### User Story
As an Volunteer, I want to view basic details for a posting, in order to decide whether I want to sign up for, edit times, or view more details of a posting.

### Solution (Core Elements)
A volunteer posting card includes the following details.

- Role title/event name
- Shift date/time
- List relevant skills
- Deadline to submit

If the volunteer wants to sign up for a posting, they can click the 'Submit availability' button.
If the volunteer wants more info on the posting, they can click the 'View details' button.

Note that events have a set date and time (see first image below), while volunteer opportunities have a date range and 'See posting details' for time (see second image below).

### Mockups

Other designs: https://www.figma.com/file/KWaJRNJcOiW8ThZJ8Sl6On/Volunteer-Experience?node-id=157%3A3123

## Technical
Create a new component called `PostingCard.tsx` in `frontend\src\components\volunteer\posting`. 

## Dev Notes

**Chakra docs for recommended components:**
* [Badge](https://chakra-ui.com/docs/data-display/badge)
* [Divider](https://chakra-ui.com/docs/data-display/divider)
* [Box](https://chakra-ui.com/docs/layout/box)
* [Icon](https://chakra-ui.com/docs/media-and-icons/icon)

### PR DESCRIPTION
**Pull request by Boyazhang1**
## Ticket link

Closes #82 (https://app.zenhub.com/workspaces/sistering-614f8dad6afeb8001d428787/issues/uwblueprint/sistering/82)

## Implementation description
- created a volunteer posting card component that is used in the volunteer postings page, based on the Figma design
Design
Screen Shot 2022-02-06 at 3 26 37 PM

Implementation
Screen Shot 2022-02-06 at 1 30 08 PM

## Steps to test

In VolunteerPostingsPage.tsx, filter the postingArr (which consists of sample data) to see how the volunteer posting cards are displayed. (note the volunteer postings page is still in progress)
Expected results: 
Screen Shot 2022-02-06 at 3 24 13 PM

## What should reviewers focus on?
- implementation matches design, chakra-ui components are being used correctly 
- prop type and values are correct
- whether types are correct/appropriate

## Checklist
- [x ] My PR name is descriptive and in imperative tense
- [ x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ x] I have run the appropriate linter(s)
- [ x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by sherryhli — CHANGES_REQUESTED** [review]
Great work so far! 😄 A lot of my comments relate to styling, although I realize that the designs may have changed between the time you started working on this ticket and now. Sorry for adding these extra requirements! However, it'd be nice if we can get these changes in now so that we're making use of the theming system

Please also resolve the lint + formatter warnings by first running `yarn fix` in the frontend, and then manually addressing the remaining issues
```
docker exec -it sistering_frontend /bin/bash -c "yarn fix"
```

Note that I did not review the posting page as that will be merged in a separate PR. Feel free to keep it for now to simplify testing, but please remove it from this branch prior to merging

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+              weekday: "long",
+              month: "long",
+              day: "2-digit",
+              timeZone: "UTC",
+            })} - ${endDate.toLocaleDateString("en-US", {
+              weekday: "long",
+              month: "long",
+              day: "2-digit",
+              timeZone: "UTC",
+            })}`}
```
Comment (LenaNguyen) [inline frontend/src/components/volunteer/PostingCard.tsx:71]: you could use `formatDateString` in `frontend\src\utils\DateUtils.ts`

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        <Divider mb={1} />
+        <HStack justifyContent="space-between" w="100%">
+          <Text textStyle="caption" color="text.gray">
+            Deadline:{" "}
+            {autoClosingDate.toLocaleDateString("en-US", {
+              weekday: "long",
+              month: "long",
+              day: "2-digit",
+              timeZone: "UTC",
+            })}
```
Comment (LenaNguyen) [inline frontend/src/components/volunteer/PostingCard.tsx:94]: `formatDateString` in `frontend\src\utils\DateUtils.ts`

File: frontend/src/theme/index.ts [hunk 3]
```diff
@@ -122,6 +122,22 @@ const customTheme = extendTheme(
...
+            },
+          },
+          outline: {
+            container: {
+              color: "text.gray",
+              boxShadow: `inset 0 0 0px 1px ${colors.text.gray}`,
+              borderRadius: "full",
+            },
+          },
+        },
```
Comment (LenaNguyen) [inline frontend/src/theme/index.ts:140]: hmm this is already merged in `main` . Not sure why they're showing up as changes here. Can you try rebasing?

**Review by LenaNguyen — COMMENTED** [review]
Thought I'd jump in here to let you know about the util that Albert already made for formatting dates :)

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+}: PostingCardProps): JSX.Element => {
+  return (
+    <Box
+      bg="white"
+      borderRadius="8px"
+      border="2px solid"
+      borderColor="background.dark"
+      id={id}
+    >
+      <VStack p={6} align="start" spacing="12px">
```
Comment (sherryhli) [inline frontend/src/components/volunteer/PostingCard.tsx:47]: Padding should be 36px for top and bottom and 40px for left-right, doesn't look like this is the case with p={6}

Figma:
Untitled

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        <Text textStyle="heading">{title}</Text>
+        <HStack spacing={4}>
+          {/* toAdd: conditionally displaying time for event listings */}
+          <Text textStyle="caption" noOfLines={2}>
+            <TimeIcon w={4} pr={1} />
+            See Posting Details
+          </Text>
+          <Text textStyle="caption">
+            <CalendarIcon w={4} pr={1} />
+            {`${formatDateString(startDate)} - ${formatDateString(endDate)}`}
```
Comment (sherryhli) [inline frontend/src/components/volunteer/PostingCard.tsx:58]: Hmm there's actually a difference between the format produced by the util function and what's expected for this view (missing year). Can you please create a new helper function in DateUtil.ts (maybe something called `formatDateStringWithYear`) and use that instead?

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        </Text>
+        <HStack>
+          <Text textStyle="body-regular">Skills: </Text>
+          {skills.slice(0, 5).map((skill) => (
+            <Tag variant="outline" key={skill.id}>
+              {skill.name}
+            </Tag>
+          ))}
+        </HStack>
+        <Divider mb={1} />
```
Comment (sherryhli) [inline frontend/src/components/volunteer/PostingCard.tsx:72]: I think there should be a margin of 16px above and below the divider

File: frontend/src/components/volunteer/PostingCard.tsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        <HStack justifyContent="space-between" w="100%">
+          <Text textStyle="caption" color="text.gray">
+            Deadline: {formatDateString(autoClosingDate)}
+          </Text>
+          <ButtonGroup spacing="4" size="md">
```
Comment (sherryhli) [inline frontend/src/components/volunteer/PostingCard.tsx:77]: Spacing a bit too wide, should be 10px

**Review by sherryhli — CHANGES_REQUESTED** [review]
Apologies for the delay in reviewing, just a couple minor things left!

There looks to be a lot of changes from other PRs that are showing up in the diff, can you make sure to only include your own changes? Your branch is up to date with `main` so I'm not sure what exactly is happening here, so perhaps a simpler fix is to create a new branch with only the `PostingCard.tsx` file (and any other dependent changes) and make a new PR. You can briefly include the browse posting page for testing but we'll remove it prior to merging.

**Comment by sherryhli** [comment]
@Boyazhang1 as #149 is also ready for review, feel free to move the PostingCard changes over and I can review both together 🙂

**Pull request closed without merging**